.maintain/ensure-deps.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+# The script is meant to check if the rules regarding packages
+# dependencies are satisfied.
+# The general format is:
+# [top-lvl-dir] MESSAGE/[other-top-dir]
+
+# For instance no crate within `./client` directory
+# is allowed to import any crate with a directory path containing `frame`.
+# Such rule is just: `client crates must not depend on anything in /frame`.
+
+# The script should be run from the main repo directory!
+
+set -u
+
+# HARD FAILING
+MUST_NOT=(
+	"client crates must not depend on anything in /frame"
+	"client crates must not depend on anything in /node"
+	"frame crates must not depend on anything in /node"
+	"frame crates must not depend on anything in /client"
+	"primitives crates must not depend on anything in /frame"
+)
+
+# ONLY DISPLAYED, script still succeeds
+PLEASE_DONT=(
+	"primitives crates should not depend on anything in /client"
+)
+
+VIOLATIONS=()
+PACKAGES=()
+
+function check_rule() {
+	rule=$1
+	from=$(echo $rule | cut -f1 -d\ )
+	to=$(echo $rule | cut -f2 -d\/)
+
+	cd $from
```

Comment:
we have different folder structure with substrate.

so I double this will work for client, frame and primitives, and frame and primitives have already been moved to darwinia-common

Reply:
The `$to` from `$rule` is fixed, and the `$from` part is not fixed, because we have no rule run into this at this moment.  Maybe it is a good issue for gitcoin bounty.